[DOCS] Adds information about version compatibility headers (#77096)
elastic/elasticsearch: docs/reference/api-conventions.asciidoc

should also be passed with a media type value that indicates the format
of the source, such as `application/json`.

[discrete]
[[api-compatibility]]
=== REST API version compatibility

Major version upgrades often include a number of breaking changes 
that impact how you interact with {es}. 
While we recommend that you monitor the deprecation log and 
update applications before upgrading {es},
having to coordinate the necessary changes can be an impediment to upgrading. 

> To enable an application to continue to function without modification after an upgrade, 

Reviewer: I think we should be more clear the functionality (semantics) are not necessarily maintained, the compatibility is with the syntax of rest request/response.

should also be passed with a media type value that indicates the format
of the source, such as `application/json`.

[discrete]
[[api-compatibility]]
=== REST API version compatibility

Major version upgrades often include a number of breaking changes 
that impact how you interact with {es}. 

> While we recommend that you monitor the deprecation log and 

Reviewer 2: Is there a single deprecation log, or are there multiple log files? Honest question, because I'm not sure 😄

Author: Good catch. There are, in fact multiple log files: "By default, Elasticsearch rolls and compresses deprecation logs at 1GB. The default configuration preserves a maximum of five log files: four rolled logs and an active log."

Author: ```suggestion
While we recommend that you monitor the deprecation logs and 
```

update applications before upgrading {es},

> having to coordinate the necessary changes can be an impediment to upgrading. 

Reviewer 2: ```suggestion
implementing the necessary changes can be an impediment to upgrading. 
```

Reviewer 2: "having to coordinate" feels imprecise, thought that might be the intention given that we're talking about all the things that need to happen for an upgrade.

Author: It was deliberate. Using the compatibility headers enables you to decouple the implementation and deployment of app updates from the upgrade process.

To enable an application to continue to function without modification after an upgrade, 
you can include API compatibility headers that tell {es} that 
you are still using the previous version of the REST API. 

> Using the headers allows the structure of requests and responses to remain the same,

Author: ```suggestion
Using the headers allows the structure of requests and responses to remain the same;
```

> To enable an application to continue to function without modification after an upgrade, 
> you can include API compatibility headers that tell {es} that 
> you are still using the previous version of the REST API. 
> Using the headers allows the structure of requests and responses to remain the same;
> it does not guarantee the same behavior. 

Reviewer 2: ```suggestion
You can enable an existing application to function without modification after
an upgrade by including API compatibility headers, which tell {es} you are still
using the previous version of the REST API. Using these headers allows the 
structure of requests and responses to remain the same; it does not guarantee
the same behavior. 
```

Reviewer 2: I had to re-read the first sentence a few times to parse it. Could just be me, but I offered an alternative wording that might simplify readability.
